app/services/billable_metrics/aggregations/unique_count_service.rb

```diff
@@ -59,7 +60,14 @@ def running_total(options)
         (1..result.aggregation).to_a
       end
 
-      private
+      def get_previous_event_in_interval(from_datetime:, to_datetime:)
+        @from_datetime = from_datetime
```

Comment:
Just a question: I'm wondering if there is a risk at changing instance variable here since it seems `previous_event` is a cached variable.

Also it seems the method could be shared in the base_service

Reply:
you are right, moved to base service since this is shared method. 👍 
`previous_event` method relies on instance variables (from_datetime and to_datetime). Since `prorated` aggregation services and `full` aggregation services don't share same parent I have to set these instance variables for `full` aggregation service (basically the same principle as we do in `aggregate` method). I would say it's safe since we just passed `from_datetime` and `to_datetime` to full aggregation service in order to get previous_event in correct interval, these values have not changed along the way. In any aggregation service we just set these values when we call aggregate and these values are not changed for one instance of aggregation service